test/ExportQueryExecutionTreesTest.cpp

```diff
@@ -1129,244 +1128,110 @@ TEST(ExportQueryExecutionTrees, getIdTablesMirrorsGenerator) {
 // _____________________________________________________________________________
 TEST(ExportQueryExecutionTrees, ensureCorrectSlicingOfSingleIdTable) {
   auto tableGenerator = []() -> cppcoro::generator<IdTable> {
-    IdTable idTable1{1, ad_utility::makeUnlimitedAllocator<Id>()};
-    idTable1.push_back({Id::makeFromInt(1)});
-    idTable1.push_back({Id::makeFromInt(2)});
-    idTable1.push_back({Id::makeFromInt(3)});
-
-    co_yield std::move(idTable1);
+    IdTable idTable1 = makeIdTableFromVector({{1}, {2}, {3}});
+    co_yield idTable1;
   }();
 
   Result result{std::move(tableGenerator), {}, LocalVocab{}};
   auto generator = ExportQueryExecutionTrees::getRowIndices(
       LimitOffsetClause{._limit = 1, ._offset = 1}, result);
 
-  auto iterator = generator.begin();
-  ASSERT_NE(iterator, generator.end());
-
-  auto range = iterator->view_;
-  auto rangeIterator = range.begin();
-  ASSERT_NE(rangeIterator, range.end());
-  EXPECT_EQ(iterator->idTable_.at(*rangeIterator)[0], Id::makeFromInt(2));
-
-  ++rangeIterator;
-  EXPECT_EQ(rangeIterator, range.end());
-
-  ++iterator;
-  EXPECT_EQ(iterator, generator.end());
+  auto referenceTable = makeIdTableFromVector({{2}});
+  EXPECT_THAT(convertToVector(std::move(generator)),
+              ElementsAre(Eq(std::cref(referenceTable))));
```

Comment:
From https://google.github.io/googletest/reference/matchers.html 

> **WARNING**: Equality matching via EXPECT_THAT(actual_value, expected_value) is supported, however note that implicit conversions can cause surprising results. For example, EXPECT_THAT(some_bool, "some string") will compile and may pass unintentionally.

Also the `Eq` seems to be required for the cref stuff to work otherwise it tries to copy the `IdTable`

Reply:
so you say the `cref(...)` without `Eq()` doesn't also work?

Reply:
Exactly. No idea why but I just tried it to confirm